### PR TITLE
Fix progressbar and notifications segfaults in test

### PR DESCRIPTION
**Pull request by Czaki**
# Description

Today morning I notice that every time tests fail I see black "plugin.py" windows which are closed exactly at the same time when StopIteration was printed to console. When running tests without `test_progress` these windows do not show. 

Zrzut ekranu z 2021-05-17 08-51-15

Then in this PR, I try to force its closing. And it works locally. 

So it looks like segfault in `MIN_REQ` is connected with the pending notification dialog.
I think that it could be connected with the same reason why the last test was disabled on python 3.7. 

https://github.com/napari/napari/blob/419cfe0a759ae1d3c7fb2747f9ab149390a43ee8/napari/_qt/_tests/test_qt_notifications.py#L125-L135

This PR is a little brutal but may be an inspiration for a person who better understands notification code or may be used as a temporary workaround for release 0.4.8 if no one has some idea for a better solution.  If using please check the mechanism of cleaning the`_instances` class attribute of `NapariQtNotification` 

cc: @tlambert03 @sofroniewn 

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Comment by Czaki** [comment]
It looks now good, but I still think that it could be done better.

**Comment by Czaki** [comment]
I can not reproduce fails locally. Could anyone else try?

**Comment by tlambert03** [comment]
I'm surprised by this one.  When the dialog is created, it is parented to something in the widget parent chain, and so shouldn't need to be cleaned up explicitly like this... (it would leak a widget if it wasn't).  Can you expand on the reasoning here?

**Comment by Czaki** [comment]
> I'm surprised by this one. When the dialog is created, it is parented to something in the widget parent chain, and so shouldn't need to be cleaned up explicitly like this... (it would leak a widget if it wasn't). Can you expand on the reasoning here?

I think that there is no parent. There is no Napari Viewer instance so there is no parent:

https://github.com/napari/napari/blob/419cfe0a759ae1d3c7fb2747f9ab149390a43ee8/napari/_qt/dialogs/qt_notification.py#L95-L99

**Comment by tlambert03** [comment]
ahhh, yep for that test... that makes sense.  thanks

**Comment by Czaki** [comment]
Feel free to push anything to this branch.

**Comment by Czaki** [comment]
I recall that sometimes adding to `qtbot` without show widget does not work so I add a small fix but have no next idea.

**Review by sofroniewn — APPROVED** [review]
This looks good to me, all tests are passing. But I'll wait for @tlambert03 to approve too before merge

**Review by tlambert03 — APPROVED** [review]

**Pull request merged**
